SchedulerCreator/src/js/exporters/HTML/HTMLExporter.js

```diff
@@ -34,6 +35,18 @@ module.exports = (timedScheduleObject) => {
                     if (theClass.teacher) {
                         classCode += `<span class="teacher">${theClass.teacher}</span>`;
                     }
+                    if (theClass.color) {
+                        if(colors[i]) {
+                            document.getElementsByClassName("class").style.backgroundColor = colors
+                        } else {
+                            let color = []
+                            for (_= 0;_<3;_++) {
+                                color.push(Math.floor(Math.random() * 255))
+                            }
+                            document.getElementsByClassName("class").style.backgroundColor = "rgba(" + color[0] + "," + color[1] + "," + color[2] + ", .5)"
+                            // In the case that all of the colors are used, the program would generate a new color with a light opacity in order for the class to be readable
```

Comment:
We know there is a max amount of classes possible (6 classes a day times 8 days = 48 colors). I suggest we create a bank of 48 colors.

Reply:
Hard to do when the colors get really similar

Reply:
You really only need like 10-12 solid colors, after that in can be same base color just a different shade as these will rarely ever come up.